src/commands/sync.ts

```diff
@@ -434,6 +434,8 @@ export default class Sync extends BaseCommand<typeof Sync> {
                     await fs.ensureDir(path.dirname(filepath), { mode: 0o755 });
                     if (!file.path.endsWith("/")) {
                       await fs.writeFile(filepath, Buffer.from(file.content, file.encoding ?? FileSyncEncoding.Utf8), { mode: file.mode });
+                    } else {
+                      await fs.ensureDir(filepath, { mode: 0o755 });
```

Comment:
We explicitly set directory mode to `0o755` above for files in directories so we should probably do the same here